### PR TITLE
Fix metadata not beeing sent to AWS

### LINKED ISSUE #669: Metadata is not added to pre-signed PutObjectRequest
Doesn't seem to support the addition of Metadata.

This is my code below but the result doesn't add the correct headers to the pre-signed request.
```
        $input = new PutObjectRequest([
            'Bucket' => $this->configuration->getExerciseVideoBucket()
            'Key' => $remoteFilename,
            'Metadata' => [
                "x-amz-meta-filename: $filename"
            ]
        ]);

       $url = $this->s3->presign($input, new \DateTimeImmutable('+20 min'));

```

### PR DESCRIPTION
**Pull request by jderusse**
fixes #669

### CODE REVIEW

File: src/Service/S3/tests/Unit/Result/GetObjectOutputTest.php [hunk 1]
```diff
@@ -57,7 +57,8 @@ public function testMetadata()
 
         $metadata = $result->getMetadata();
         self::assertCount(1, $metadata);
-        self::assertArrayHasKey('x-amz-meta-tobias', $metadata);
-        self::assertEquals('nyholm', $metadata['x-amz-meta-tobias']);
+        self::assertArrayNotHasKey('x-amz-meta-tobias', $metadata);
+        self::assertArrayHasKey('tobias', $metadata);
```
Comment (jderusse) [inline src/Service/S3/tests/Unit/Result/GetObjectOutputTest.php:61]: This change mimics AWS SDK

IMHO `x-amz-meta-` is an internal prefix and shoudn't have leaked to end user. When people use the official SDK to send `tobias`, they expect to get `tobias` in return.

Comment (Nyholm) [reply src/Service/S3/tests/Unit/Result/GetObjectOutputTest.php:61]: Yes, True. I missed this before. 

Thank you

File: src/Service/S3/src/Result/GetObjectOutput.php [hunk 2]
```diff
@@ -454,7 +454,7 @@ protected function populateResult(Response $response): void
         $this->Metadata = [];
         foreach ($headers as $name => $value) {
             if ('x-amz-meta-' === substr($name, 0, 11)) {
-                $this->Metadata[$name] = $value[0];
+                $this->Metadata[substr($name, 11)] = $value[0];
```
Comment (Nyholm) [inline src/Service/S3/src/Result/GetObjectOutput.php:457]: Could the user manually add metadata in the S3 dashboard. Would that be prefixed here?

Comment (jderusse) [reply src/Service/S3/src/Result/GetObjectOutput.php:457]: I'm not sure to understand what you mean. 

note: Aws SDK removes the prefix as well

Comment (Nyholm) [reply src/Service/S3/src/Result/GetObjectOutput.php:457]: I meant: Will `$name` always include the prefix? No matter how the metadata was added?

Comment (jderusse) [reply src/Service/S3/src/Result/GetObjectOutput.php:457]: prefix it checked at line 456

Comment (Nyholm) [reply src/Service/S3/src/Result/GetObjectOutput.php:457]: Oh, sorry. I didnt see that.

**Comment by Nyholm** [comment]
Thank you

**Comment by phamviet** [comment]
Thanks for this PR. May i know when this get released? Currently i have to use 1.2-dev to get this fix but would be good to have it stable instead.

**Comment by Nyholm** [comment]
Thank you for the ping. We've must have forgotten this. I've prepared a release here: https://github.com/async-aws/aws/pull/734

**Comment by phamviet** [comment]
Thanks @Nyholm! Did not expect it happens so quickly. Very appreciate your work!

**Pull request merged**